src/pb_tensor.cc

```diff
@@ -305,11 +339,30 @@ PbTensor::Memory()
 
 #ifdef TRITON_PB_STUB
 std::shared_ptr<PbTensor>
-PbTensor::FromDLPack(const std::string& name, const py::capsule& dlpack_tensor)
+PbTensor::FromDLPack(const std::string& name, const py::object& tensor)
 {
   if (name == "") {
     throw PythonBackendException("Tensor name cannot be an empty string.");
   }
+  if(py::isinstance<py::capsule>(tensor)){
+    return FromDLPackCapsule(name, tensor);
+  } else if (py::hasattr(tensor, "__dlpack__")) {
+    // Python backend does not support async executions on PbTensor objects,
+    // thus we can count on the default `stream=None` argument.
+    // For CPU, `stream=None` is the only accepted argument 
+    // according to array API. For GPU, when `stream=None`  producer must 
+    // assume the legacy default stream.
+    // Reference:
+    // https://data-apis.org/array-api/latest/API_specification/generated/array_api.array.__dlpack__.html
+    return FromDLPackCapsule(name, tensor.attr("__dlpack__")());
+  }else{
+    throw PythonBackendException("Provided tensor is not supported.");
```

Comment:
Perhaps mention that the tensor must be dlpack capsule or have a `__dlpack__` attribute.

Reply:
Added in Revision 1